Multiple output formats fails using Linux MSBuild

This isn't a problem with Coverlet but maybe something to add to the documentation.

If you want to specify multiple output formats, the docs say you need to escape the quotes around the formats

`/p:CoverletOutputFormat=\"json,cobertura\"`

so in MSBuild it'd look like

```xml
<Exec Command="dotnet test MySolution.sln -c $(Configuration) /p:CollectCoverage=true /p:CoverletOutput=log /p:MergeWith=log/coverage.json /p:CoverletOutputFormat=\&quot;json,cobertura\&quot; -m:1" />
```

Unfortunately, [there's an issue with Linux MSBuild](https://github.com/microsoft/msbuild/issues/3468) that translates escape backslashes `\` to slashes `/` on the fly. Thus on, say Azure DevOps where you're running on an Ubuntu agent, you'll get an error like:

```text
MSBUILD : error MSB1006: Property is not valid. [/home/vsts/work/1/s/default.proj]
  Switch: cobertura/
```
The quotes aren't escaped and the command line gets parsed wrong.

On Windows this works as expected, and if you use `dotnet msbuild` on Linux it also works as expected. You'll see the error if you're on Linux and:

- You call the binary Linux MSBuild rather than `dotnet msbuild` OR
- You're in Azure DevOps on a Linux agent and use the `MSBuild` task (which uses the binary Linux MSBuild)

And maybe this issue is enough, something folks can search for and find later. I just know I spent like two hours trying to figure out why things worked fine on my Mac using `dotnet msbuild` but the Ubuntu agent was failing every time.

--- a/Documentation/MSBuildIntegration.md
+++ b/Documentation/MSBuildIntegration.md
@@ -74,7 +74,7 @@ With Coverlet you can combine the output of multiple coverage runs into a single
 dotnet test /p:CollectCoverage=true /p:MergeWith='/path/to/result.json'
 ```
 
-The value given to `/p:MergeWith` **must** be a path to Coverlet's own json result format. The results in `result.json` will be read, and added to the new results written to by Coverlet.  
+The value given to `/p:MergeWith` **must** be a path to Coverlet's own json result format. The results in `result.json` will be read, and added to the new results written to by Coverlet.
 [Check the sample](Examples.md).
 
 ## Threshold
@@ -167,6 +167,19 @@ VSTS builds do not require double quotes to be unescaped:
 ```
 dotnet test --configuration $(buildConfiguration) --no-build /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura /p:CoverletOutput=$(Build.SourcesDirectory)/TestResults/Coverage/ /p:Exclude="[MyAppName.DebugHost]*%2c[MyAppNamet.WebHost]*%2c[MyAppName.App]*"
 ```
+
+## Note for Linux users
+
+[There is an issue with MSBuild on Linux](https://github.com/microsoft/msbuild/issues/3468) affects the ability to escape quotes while specifying multiple comma-separated values. Linux MSBuild automatically translates `\` to `/` in properties, tasks, etc. before using them, which means if you specified `/p:CoverletOutputFormat=\"json,opencover\"` in an MSBuild script, it will be converted to `/p:CoverletOutputFormat=/"json,opencover/"` before execution. This yields an error similar to the following:
+
+```text
+MSBUILD : error MSB1006: Property is not valid. [/home/vsts/work/1/s/default.proj]
+  Switch: opencover/
+```
+
+You'll see this if directly consuming Linux MSBuild or if using the Azure DevOps `MSBuild` task on a Linux agent.
+
+The workaround is to use the .NET Core `dotnet msbuild` command instead of using MSBuild directly. The issue is not present in `dotnet msbuild` and the script will run with correctly escaped quotes.
 
 ## SourceLink
 

